### PR TITLE
fix #3094

### LINKED ISSUE #3094: Widget for updatable image
I don't want to reinvent the wheel, so will check before.
I use GenPollText for my keyboard layout indicator but instead I want to see the flag (image). As I change layout the image should be changed. Do Qtile has such widget or proper way to do that?

### PR DESCRIPTION
**Pull request by zordsdavini**
Added to Image widget update functionality. Should fix #3094 issue

### CODE REVIEW

File: libqtile/widget/image.py [hunk 1]
```diff
@@ -94,3 +97,8 @@ def calculate_length(self):
             return self.img.width + (self.margin_x * 2)
         else:
             return self.img.height + (self.margin_y * 2)
+
+    def cmd_update(self, filename):
+        self.filename = filename
+        self._update_image()
+        self.draw()
```
Comment (elParaguayo) [inline libqtile/widget/image.py:104]: If your new image is not the same size as the previous one then you should do a `self.bar.draw()` to ensure the other widgets' positions are also adjusted.

Comment (zordsdavini) [reply libqtile/widget/image.py]: added update for length check and redraw but got some check issue what I don't understand

Comment (elParaguayo) [reply libqtile/widget/image.py]: Those check issues aren't caused by this. Still trying to track that one down...

Comment (elParaguayo) [reply libqtile/widget/image.py]: If you rebase this branch onto the latest master then those failing tests should disappear.

Comment (zordsdavini) [reply libqtile/widget/image.py]: rebased

**Comment by m-col** [comment]
Works great, thanks! Could you add a line in the changelog so that others can find the new feature?

**Comment by zordsdavini** [comment]
@m-col added info to CHANGELOG

**Comment by m-col** [comment]
Great, thanks, auto-merge is queued!

**Comment by zordsdavini** [comment]
@m-col automatical merge failed because last time I did merge with master (old habit). Did rebase

**Pull request merged**